draft-ietf-quic-transport.md

```diff
@@ -1056,18 +1057,20 @@ An endpoint might need to stop accepting previously issued connection IDs in
 certain circumstances.  Such an endpoint can cause its peer to retire connection
 IDs by sending a NEW_CONNECTION_ID frame with an increased Retire Prior To
 field.  The endpoint SHOULD continue to accept the previously issued connection
-IDs until they are retired by the peer.  If the endpoint can no longer process
-the indicated connection IDs, it MAY close the connection.
+IDs until a connection ID greater than or equal to the Retire Prior To value is
+used by the peer.  If the endpoint can no longer process the indicated
```

Comment:
"until a connection ID greater than or equal to the Retire Prior To value" doesn't work.

Say you have 10 active.  You can still issue up to 15  before you send "Retire Prior To" with a value of 10.  So use of connection ID 15 is no good signal that the retirement was accepted.

I think that what you meant to say here is "until the peer uses a connection ID with a sequence number greater than the one included in the NEW_CONNECTION_ID that invalidated the previous connection ID" or some such.  Noting of course that you can send NEW_CONNECTION with seq 16 and RPT of 0 followed by NEW_CONNECTION_ID { seq 16, RPT 10 } totally legitimately; in which case you can't rely on use of CID[16] to mean anything.

It would also not be entirely wise to drop connection IDs at the point that you see the new ones being used, even if your assumption was true.  Though you might have confirmation that a stateless reset won't kill the connection, you would still be dropping perfectly good packets that are still totally usable.

Reply:
I agree with Martin here. The next needs to be updated to refer to the NCID sequence number instead.